src/pages/cb-bank-branches/index.js

```diff
@@ -0,0 +1,172 @@
+
+const BankBranches = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    return await getRequest({
+      extension: CashBankRepository.BankBranches.qry,
+      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&filter=`
+    })
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+
+    filterBy,
+    clearFilter,
+    refetch,
+
+    access
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: CashBankRepository.BankBranches.qry,
+    datasetId: ResourceIds.BankBranches,
+    filter: {
+      filterFn: fetchWithFilter
+    }
+  })
+
+  async function fetchWithFilter({ filters, pagination }) {
+    if (filters?.qry) {
+      return await getRequest({
+        extension: CashBankRepository.BankBranches.snapshot,
+        parameters: `_filter=${filters.qry}`
+      })
+    } else {
+      return fetchGridData({ _startAt: pagination._startAt || 0, params: filters?.params })
+    }
+  }
+
+  const invalidate = useInvalidate({
+    endpointId: CashBankRepository.BankBranches.qry
+  })
+
+  const columns = [
+    {
+      field: 'name',
+      headerName: _labels.name,
+      flex: 1
+    },
+    {
+      field: 'bankName',
+      headerName: _labels.bank,
+      flex: 1
+    },
+    {
+      field: 'swiftCode',
+      headerName: _labels.swiftCode,
+      flex: 1
+    },
+    {
+      field: 'countryName',
+      headerName: _labels.country,
+      flex: 1
+    }
+  ]
+
+  const add = () => {
+    openForm()
+  }
+
+  function openForm(recordId) {
+    stack({
+      Component: BankBranchesForm,
+      props: {
+        labels: _labels,
+        recordId,
+        maxAccess: access
+      },
+      width: 780,
+      height: 550,
+      title: _labels.bankBranches
+    })
+  }
+
+  const edit = obj => {
+    openForm(obj.recordId)
+  }
+
+  const del = async obj => {
+    try {
+      await postRequest({
+        extension: CashBankRepository.BankBranches.del,
+        record: JSON.stringify(obj)
+      })
+      invalidate()
+      toast.success(platformLabels.Deleted)
+    } catch (error) {}
+  }
+
+  const onApply = ({ search, rpbParams }) => {
+    if (!search && rpbParams.length === 0) {
+      clearFilter('params')
+    } else if (!search) {
+      filterBy('params', rpbParams)
+    } else {
+      filterBy('qry', search)
+    }
+    refetch()
+  }
+
+  const onSearch = value => {
+    filterBy('qry', value)
+  }
+
+  const onClear = () => {
+    clearFilter('qry')
+  }
+
+  return (
+    <VertLayout>
+      <Fixed>
+        <RPBGridToolbar
+          onAdd={add}
+          maxAccess={access}
+          onApply={onApply}
+          onSearch={onSearch}
+          onClear={onClear}
+          reportName={'CABNB'}
```

Comment:
can't the report name be a label? or there is no need for it?

Reply:
no need for it